lib/internal/url.js

```diff
@@ -62,6 +62,12 @@ function toUSVString(val) {
   return _toUSVString(str, match.index);
 }
 
+// Returns true if `val` is of the spec Object type (including functions), but
+// false if `val` is of the spec Null type.
+function isObject(val) {
+  return (typeof val === 'object' && val !== null) || typeof val === 'function';
```

Comment:
If we do want a helper function..this seems to be equivalent to `!util.isPrimitive(val)`?